python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -612,7 +617,7 @@ def determine_asset_partitions_to_reconcile_for_freshness(
 
             # the set of asset keys which are involved in some constraint and are actually upstream
             # of this asset
-            relevant_upstream_keys = frozenset(
+            relevant_upstream_keys = set(
```

Comment:
frozenset gets surprisingly expensive, as hashing requires sorting the elements! this was just a tiny change, but we use frozenset elsewhere, so I may end up seeing if it's possible to remove those other uses in a future PR.